lib/aggregate/attribute/bitfield.rb

```diff
@@ -34,4 +43,27 @@ def validation_errors(_value)
   def klass
     @klass ||= options[:limit] ? Aggregate::Bitfield.limit(options[:limit]) : Aggregate::Bitfield
   end
+
+  def new_klass(value)
+    klass.new(value, mapping: mapping, default: default_value)
+  end
```

Comment:
It is a little weird to me that every instance of the bitfield needs to be constructed with the mapping and default value.  These are class concerns not instance concerns.  With limit, we were able to create a new class that represented the configuration and then the instance code was simpler and was not copying around a bunch of unneeded attributes. Maybe we can spend a couple minutes trying to figure out how this can be done? (The tricky bit is coming up with a name for the class, but I think we can do that with a hash...)